packages/server/lib/controllers/oauth.controller.ts

```diff
@@ -1006,7 +1006,20 @@ class OAuthController {
                 return publisher.notifyErr(res, channel, providerConfigKey, connectionId, WSErrBuilder.UnknownError());
             }
 
-            let connectionConfig = { ...session.connectionConfig, ...tokenMetadata, ...callbackMetadata };
+            let connectionConfig = {
```

Comment:
This will ensure that users `connecionConfig` takes precedence, since in connect-ui, we have the `optional` field, this makes the users `connectionConfig` an empty string when applied, if the empty string is present in the metadata, we override it, else we use user's `connectionConfg`